### DISCUSSION TITLE
Should we include the recipient in the `StreamLD` `StreamLL` and `StreamLT`?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#849] [Ideas]
Since the `getStream` no longer returns the exact struct from storage, but a built one at the execution time: `Lockup.Stream` + unique information depending the Lockup model (e.g. segments for dynamic).

I think now, it would be better to also include the `recipient`, so that this natspec is actually correct 😅

https://github.com/sablier-labs/v2-core/blob/cbe1c0d61d74217d6ad01599318d5551f3381b2b/src/types/DataTypes.sol#L180-L181

The improvement would be:
```solidity
LockupLinear.StreamLL memory streamDetails = lockupLinear.getStream(streamId);
address recipient = lockupLinear.getRecipient(streamId);
```

vs

```solidity
LockupLinear.StreamLL memory streamDetails = lockupLinear.getStream(streamId); // includes the recipient
```

@PaulRBerg @smol-ninja curious about what you guys have to say.

**Comment by andreivladbrg** [comment]
Another idea would be to rename `getStream` to `getStreamDetails`, tho this change might require more work and we are pressed by time.

**Reply by PaulRBerg** [reply]
I don't like this approach

**Comment by PaulRBerg** [comment]
Good catch, @andreivladbrg.

I am just now realizing that the struct returned by `getStream` has never included the recipient. Like never ever. That's funny.

I'm fully on board with this proposal.

**Reply by andreivladbrg** [reply]
great, thanks

**Comment by smol-ninja** [comment]
I like this idea, Andrei. It would make the get API implementation much cleaner. I also prefer `getStream` over `getStreamDetails`, `getStream` in itself conveys that it would fetch details of the stream, so I think there is no need to specify the keyword "Details" specifically.

**Comment by andreivladbrg** [comment]
Merged https://github.com/sablier-labs/v2-core/pull/850

**Reply by smol-ninja** [reply]
We also need to update some tests in periphery wherever `getStream` is used.

**Reply by smol-ninja** [reply]
I've made the changes since the test wouldn't pass without it: https://github.com/sablier-labs/v2-periphery/pull/304

**Reply by PaulRBerg** [reply]
can't wait to move to [package tethering](https://github.com/sablier-labs/v2-core/issues/808) and merge V2 Core and V2 Periphery into a long-awaited Lockup repository

**Reply by smol-ninja** [reply]
Me neither. I have realized how much time we spend in managing two repos when they could just have been one. Recurring commits like "update bun lockfile" and dependencies management are just 😤
